packages/orbit-components/src/Checkbox/index.tsx

```diff
@@ -239,25 +50,56 @@ const Checkbox = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
         onChange={onChange}
         ref={ref}
         readOnly={readOnly}
-        error={hasError}
       />
       {cloneWithTooltip(
         tooltip,
-        <StyledIconContainer
-          disabled={disabled}
-          checked={checked}
-          onClick={readOnly ? preventOnClick : undefined}
+        <div
+          className={cx(
+            "orbit-checkbox-icon-container",
+            "relative box-border",
+            "border border-solid",
+            "flex shrink-0 grow-0 basis-auto items-center justify-center",
+            "h-icon-medium w-icon-medium",
+            "rounded-large de:rounded-normal",
+            "duration-fast transition-all ease-in-out",
+            "peer-focus:outline-blue-normal peer-focus:outline peer-focus:outline-2",
+            "[&_>_svg]:w-icon-small [&_>_svg]:h-icon-small [&_>_svg]:invisible peer-checked:[&_>_svg]:visible",
+            "[&_>_svg]:flex [&_>_svg]:items-center [&_>_svg]:justify-center",
+            "active:scale-95",
+            disabled
```

Comment:
could we pls not do nested ternaries & nested arrays? it is very difficult to visually parse 😕 by the time I get into the deepest conditional structure I have no idea what conditions were already applied

Reply:
resolved ;)